sklearn/utils/tests/test_stats.py

```diff
@@ -13,55 +14,43 @@ def test_weighted_percentile():
     y[50] = 1
     sw = np.ones(102, dtype=np.float64)
     sw[-1] = 0.0
-    score = _weighted_percentile(y, sw, 50)
-    assert approx(score) == 1
+    value = _weighted_percentile(y, sw, 50)
+    assert approx(value) == 1
 
 
 def test_weighted_percentile_equal():
     y = np.empty(102, dtype=np.float64)
     y.fill(0.0)
     sw = np.ones(102, dtype=np.float64)
     sw[-1] = 0.0
-    score = _weighted_percentile(y, sw, 50)
-    assert score == 0
+    value = _weighted_percentile(y, sw, 50)
+    assert value == 0
 
 
 def test_weighted_percentile_zero_weight():
     y = np.empty(102, dtype=np.float64)
     y.fill(1.0)
     sw = np.ones(102, dtype=np.float64)
     sw.fill(0.0)
-    score = _weighted_percentile(y, sw, 50)
-    assert approx(score) == 1.0
+    value = _weighted_percentile(y, sw, 50)
+    assert approx(value) == 1.0
 
 
 def test_weighted_percentile_zero_weight_zero_percentile():
     y = np.array([0, 1, 2, 3, 4, 5])
     sw = np.array([0, 0, 1, 1, 1, 0])
-    score = _weighted_percentile(y, sw, 0)
-    assert approx(score) == 2
+    value = _weighted_percentile(y, sw, 0)
+    assert approx(value) == 2
 
-    score = _weighted_percentile(y, sw, 50)
-    assert approx(score) == 3
+    value = _weighted_percentile(y, sw, 50)
+    assert approx(value) == 3
 
-    score = _weighted_percentile(y, sw, 100)
-    assert approx(score) == 4
-
-
-def test_weighted_median_equal_weights():
```

Comment:
`test_weighted_median_equal_weights()` was checking if `_weighted_percentile` and `np.median` (both with `sample_weights` all ones and probability level=0.5) would return the same percentiles, but accidentally checking it on an odd number of samples (as opposed to even numbers). With an odd number of samples, `_weighted_percentile` with the default arguments would always fall on a precise value and not on the next lower one (see discussion in #17370).

I would therefore suggest to remove this test here, as it is not testing the correct thing (and setting the number of samples to an even number, it would fail as expected).

Reply:
Let's keep it but make it explicit that currently this is only expected to hold true for odd numbers of data points and link to #17370 in a comment.

If/when we implement #17370 we will expand the test to also check the equivalence with an even number of data points.

Reply:
Yes, I have worked on the test comment to make this clear. Please let me know if this is clear enough.